### PR TITLE
refactor: replace ForkChoiceStream type with generic stream type

### LINKED ISSUE #10392: Replace `ForkChoiceStream` type with generic stream type
### Describe the feature

we currently have the same thing twice
https://github.com/paradigmxyz/reth/blob/170a396903c2373d995e1289267906ef38327898/crates/chain-state/src/notifications.rs#L160-L166

https://github.com/paradigmxyz/reth/blob/170a396903c2373d995e1289267906ef38327898/crates/chain-state/src/notifications.rs#L185-L185

we can phase out the former and maybe find a more appropriate name for the latter

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jenpaff**
Closes #10392

### CODE REVIEW

File: crates/chain-state/src/notifications.rs [hunk 1]
```diff
@@ -152,54 +152,34 @@ pub trait ForkChoiceSubscriptions: Send + Sync {
...
         }
     }
 }
 
-/// A stream for block state watch channels (pending, safe or finalized watchers)
+/// A stream for fork choice watch channels (pending, safe or finalized watchers)
 #[derive(Debug)]
 #[pin_project::pin_project]
-pub struct BlockStateNotificationStream<T> {
+pub struct ForkChoiceStream<T> {
```
Comment (jenpaff) [inline crates/chain-state/src/notifications.rs:170]: I actually quite like the name ForkChoiceStream, let me know if this is appropriate for the type or needs renaming

File: crates/chain-state/src/notifications.rs [hunk 2]
```diff
@@ -152,54 +152,34 @@ pub trait ForkChoiceSubscriptions: Send + Sync {
...
-    /// Creates a new `BlockStateNotificationStream`
+impl<T: Clone + Sync + Send + 'static> ForkChoiceStream<T> {
+    /// Creates a new `ForkChoiceStream`
     pub fn new(rx: watch::Receiver<Option<T>>) -> Self {
         Self { st: WatchStream::new(rx) }
     }
 }
 
-impl<T: Clone + Sync + Send + 'static> Stream for BlockStateNotificationStream<T> {
+impl<T: Clone + Sync + Send + 'static> Stream for ForkChoiceStream<T> {
```
Comment (onbjerg) [inline crates/chain-state/src/notifications.rs:182]: nit: i slightly prefer the other `poll_next` impl, so maybe we could port that over?

**Review by onbjerg — APPROVED** [review]
lgtm, i also like the name `ForkChoiceStream`

File: crates/chain-state/src/notifications.rs (hunk 2 above)
Comment (jenpaff) [reply crates/chain-state/src/notifications.rs:182]: fixed

**Pull request merged**